src/manifest/pcd.rs

```diff
@@ -0,0 +1,610 @@
+        self.data.port_count as usize
+    }
+
+    /// Returns the total number of component RoTs connected to this RoT.
+    pub fn components(&self) -> usize {
+        self.data.component_count as usize
+    }
+
+    /// Returns this RoT's I2C address.
+    pub fn rot_addr(&self) -> u8 {
+        self.data.addr
+    }
+
+    /// Returns this RoT's MCTP endpoint ID.
+    pub fn rot_eid(&self) -> u8 {
+        self.data.eid
+    }
+
+    /// Returns the MCTP bridge's I2C address.
+    pub fn bridge_addr(&self) -> u8 {
+        self.data.bridge_addr
+    }
+
+    /// Returns the MCTP bridge's endpoint ID.
+    pub fn bridge_eid(&self) -> u8 {
+        self.data.bridge_eid
+    }
+
+    /// Returns an iterator over the `SpiFlash` elements of this PCD.
+    ///
+    /// The returned values only contain the `Toc` information for the entry,
+    /// allowing the user to lazily select which entries to read from flash.
+    pub fn spi_flashes(
+        &self,
+    ) -> impl Iterator<Item = SpiFlashEntry<'_, 'pcd, P>> {
+        self.entry()
+            .children_of(ElementType::SpiFlash.into())
+            .map(move |entry| SpiFlashEntry { rot: self, entry })
+    }
+}
+
+/// A [`SpiFlash`] element entry in a PCD's `Toc`.
+///
+/// This type allows for lazily reading the [`SpiFlash`] described by this
+/// entry, as obtained from [`SelfRot::spi_flashes()`].
+pub struct SpiFlashEntry<'a, 'pcd, Provenance = provenance::Signed> {
+    rot: &'a SelfRot<'a, 'pcd, Provenance>,
+    entry: TocEntry<'a, 'pcd, Pcd>,
+}
+
+impl<'a, 'pcd, P: Provenance> SpiFlashEntry<'a, 'pcd, P> {
+    /// Returns the `Toc` entry defining this element.
+    pub fn entry(&self) -> TocEntry<'a, 'pcd, Pcd> {
+        self.entry
+    }
+
+    /// Reads the contents of this element into memory, verifying its hash
+    /// and potentially allocating it on `arena`.
+    pub fn read(
+        self,
+        hasher: &mut dyn hash::Engine,
+        arena: &'pcd dyn Arena,
+    ) -> Result<&'pcd SpiFlash, Error> {
+        let (sf, _) = self.entry.read_with_header::<SpiFlash, P>(
+            self.rot.pcd.container.flash(),
+            arena,
+            hasher,
+        )?;
+        Ok(sf)
+    }
+}
+
+/// A flash device that can be accessed over SPI.
+#[derive(Clone, Copy, Debug, FromBytes, AsBytes)]
+#[repr(C)]
+pub struct SpiFlash {
+    port_id: u8,
+    port_flags: u8,
+    // The usage of this byte is currently not specified.
+    _policy: u8,
+    pulse_interval: u8,
+    spi_freq: u32,
+}
+
+/// A SPI flash device mode.
+#[allow(missing_docs)]
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum SpiFlashMode {
+    Dual,
+    Single,
+    DualFiltered,
+    SingleFiltered,
+}
+
+/// A SPI flash reset policy setting.
+///
+/// This setting describes what to do with a port's reset line when the flash
+/// is being verified with respect to a PFM.
+#[derive(Clone, Copy, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub enum SpiResetPolicy {
```

Comment:
I find this name a bit confusing because I read it as the SPI flash reset policy, whereas in reality it is the policy of the SPI host attached to the SPI flash device. Maybe `SpiHostDeviceResetPolicy`? Anyway, I think the docstring refers to it as a `port`. I wonder if this assumes that there is MUX or something between the host and the device...